sdk/go/pulumi/run_test.go

```diff
@@ -238,3 +240,125 @@ func TestInvoke(t *testing.T) {
 	}, WithMocks("project", "stack", mocks))
 	assert.NoError(t, err)
 }
+
+type testInstanceResource struct {
+	CustomResourceState
+}
+
+type testInstanceResourceArgs struct {
+}
+
+type testInstanceResourceInputs struct {
+}
+
+func (*testInstanceResourceInputs) ElementType() reflect.Type {
+	return reflect.TypeOf((*testInstanceResourceArgs)(nil)).Elem()
+}
+
+type testInstanceResourceInput interface {
+	Input
+
+	ToTestInstanceResourceOutput() testInstanceResourceOutput
+	ToTestInstanceResourceOutputWithContext(ctx context.Context) testInstanceResourceOutput
+}
+
+func (testInstanceResource) ElementType() reflect.Type {
```

Comment:
I think these will need to be pointer receivers. Related: https://github.com/pulumi/pulumi/pull/5916

Reply:
+1